### PR TITLE
bugfix: Fixed the toolbar menu toggler and outside click

### LINKED ISSUE #5645: Toolbar Menu Toggle and Outside Click Handling Bug
When you click on a toolbar icon, the menu that pops up isn't closing properly with a second click on the same icon. It's supposed to toggle open and closed each time you click, but right now, it just stays open. Secondly, our outside click feature, which is meant to close the menu if you click anywhere else on the screen, isn't working when you click on the user button. Ideally, clicking anywhere outside the menu should close it, but currently, clicking on the user button doesn't have this effect.

**Steps to reproduce the behavior:**

- Open the application and navigate to a page where the toolbar is present.
- Click on any toolbar icon to open the corresponding menu.
- Click on the same icon again. Observe that the menu does not close as expected.
- Now, with the menu open, click on the user button. Notice that the menu remains open instead of closing.-

**Expected behavior**
The menu should toggle (open/close) when the corresponding toolbar icon is clicked.
Any click outside of the open menu, including clicks on the user button, should close the menu.

**Screen Recording **
[sidebar open closing bug.webm](https://github.com/plone/volto/assets/138920477/997b439c-0f2c-4d37-9dce-742239a07301)

### PR DESCRIPTION
**Pull request by vikash-prem123**
This PR fixed  handling the toolbar toggle and outside click on "Personal tools".

Fix  #5645  resolve issue with toolbar menu toggler

- refactor: update handleClickOutside in Toolbar component

-   Added a simple condition to not execute the "closeMenu" in handleClickOutside function when click on the "icon" because it will execute by toggleMenu function.

**Screen Recording**

[fix-toggle-menu.webm](https://github.com/plone/volto/assets/106796672/ae2c2c31-a836-48f5-b487-547a9ba6c4d3)

@vikash-prem123

### CODE REVIEW

**Comment by stevepiercy** [comment]
@vikash-prem123 please read and follow [Contributing to Plone](https://6.docs.plone.org/contributing/index.html) and [First-time contributors](https://6.docs.plone.org/contributing/first-time.html).

**Comment by vikash-prem123** [comment]
I have carefully reviewed the [Contributing to Plone](https://6.docs.plone.org/contributing/index.html) and [First-time contributors](https://6.docs.plone.org/contributing/first-time.html) documentation. I've addressed the feedback and corrected the identified mistakes in my pull request. Could you please review the changes and provide any further feedback? Thank you.

**Comment by stevepiercy** [comment]
@vikash-prem123 you may have carefully read it, but you clearly did not follow at least two things in it, as indicated by the two failing workflows, specifically:

- [Sign and return the Plone Contributor Agreement](https://6.docs.plone.org/contributing/index.html#sign-and-return-the-plone-contributor-agreement)
- [Change log entry](https://6.docs.plone.org/contributing/index.html#change-log-entry)

Until those essential requirements are satisfied, no one will review this PR.

**Comment by stevepiercy** [comment]
@vikash-prem123 we do not review pull requests that do not pass all tests. You should do that locally before creating a pull request. See [Code quality](https://6.docs.plone.org/volto/contributing/index.html#code-quality).

We also do not guide or mentor contributors.

**Comment by vikash-prem123** [comment]
@stevepiercy  plz review my PR now.

**Comment by stevepiercy** [comment]
@vikash-prem123 what issue does this fix? You refer to this pull request, not an issue.

**Comment by vikash-prem123** [comment]
Fix #5645

File: packages/volto/news/5645.bugfix
Comment (stevepiercy) [inline packages/volto/news/5645.bugfix:1]: News items usually should make sense to both end users and developers. Implementation details are obvious when a developer reads the diff.

```suggestion
The user personal tools menu now closes when clicked a second time or when clicking outside of the component. @vikash-prem123
```

**Review by stevepiercy — CHANGES_REQUESTED** [review]
This also needs review by a core developer. In another pull request, @sneridagh referred to this https://github.com/plone/volto/blob/main/packages/volto/src/helpers/Utils/useDetectClickOutside.js. Maybe @ichim-david would know?

**Review by stevepiercy — COMMENTED** [review]
News item is fine. Still needs review by a core developer. Thank you!

**Comment by vikash-prem123** [comment]
@stevepiercy  who will merge my PR?

**Review by ichim-david — CHANGES_REQUESTED** [review]
@vikash-prem123 your fix isn't good enough, there are situations where you click on the SVG logos and target instead of being the SVG with the "icon" class you actually click on the path of the SVG so you still get the re-open

**Comment by vikash-prem123** [comment]
> @vikash-prem123 your fix isn't good enough, there are situations where you click on the SVG logos and target instead of being the SVG with the "icon" class you actually click on the path of the SVG so you still get the re-open

Ya I can see that thanks for your replay. I will fix it as soon as possible.

**Comment by vikash-prem123** [comment]
@ichim-david I have fixed that issue. please check it now.

**Review by stevepiercy — COMMENTED** [review]
I approve the news item only.

**Comment by ichim-david** [comment]
@vikash-prem123 seeing your suggested fix of modifying the user icon to add a class that you can hook on makes me see that you haven't understood the problem properly or taken my feedback into consideration as you should.
I could continue to give more things todo but then I end up fixing the issue while telling you how todo it bit by bit.

I will open a pull request with a fix and I welcome your eyes to observe what I did so that next time you do the proper testing and work so that any of the toggle buttons are fixed.

You should not rush to try and contribute to open source without having a good understanding of the product. I've worked with it for 2 years before I started to contribute and only after having real-world experience with things that require changing. 
Although everyone's path is different you need to have good experience with the product before you can make contributions and I suggest you take the time to learn it and use it before attempting to fix things.

**Comment by vikash-prem123** [comment]
@ichim-david  Thank you for your feedback. I understand the importance of gaining more experience with the product before contributing to open-source projects. I appreciate your insights and look forward to reviewing your pull request to learn from your changes. I'm open to collaboration and appreciate your guidance.

**Pull request closed without merging**
